### ISSUE TITLE
no command found udroid

### ISSUE THREAD

**Issue by TramontaG** [RandomCoderOrg/ubuntu-on-android#247]
Running the installer gives me this error. My termux can't find udroid in the path. 

![Screenshot_2024-08-05-17-56-22-699_com.termux.jpg](https://github.com/user-attachments/assets/133de98f-8dfb-4255-974e-21f4daa2bf87)

**Comment by DGxInfinitY** [comment]
Same issue.

**Comment by SaicharanKandukuri** [comment]
@TramontaG @DGxInfinitY 
Can you share full output here.

**Comment by DGxInfinitY** [comment]
> @TramontaG @DGxInfinitY Can you share full output here.

You bet. Here is the output with a fresh installation on a Google Pixel 8 Pro with Android 15.

```
$ apt update && apt upgrade -y
. <(curl -Ls https://bit.ly/udroid-installer)
Get:1 https://termux.net stable InRelease [1086 B]
Get:2 https://termux.net stable/main aarch64 Packages [229 kB]
Fetched 230 kB in 2s (116 kB/s)
11 packages can be upgraded. Run 'apt list --upgradable' to see them.
Upgrading:
  apt                curl         libc++   readline
  ca-certificates    debianutils  libcurl  termux-exec
  command-not-found  less         openssl

Summary:
  Upgrading: 11, Installing: 0, Removing: 0, Not Upgrading: 0
  Download size: 5365 kB
  Space needed: 184 kB

Get:1 https://termux.net stable/main aarch64 libc++ aarch64 27 [298 kB]
Get:2 https://termux.net stable/main aarch64 apt aarch64 2.9.7 [1060 kB]
Get:3 https://termux.net stable/main aarch64 ca-certificates all 2024.07.02 [125 kB]
Get:4 https://termux.net stable/main aarch64 openssl aarch64 3.3.1 [2320 kB]
Get:5 https://termux.net stable/main aarch64 libcurl aarch64 8.9.1 [891 kB]
Get:6 https://termux.net stable/main aarch64 curl aarch64 8.9.1 [208 kB]
Get:7 https://termux.net stable/main aarch64 less aarch64 661 [126 kB]
Get:8 https://termux.net stable/main aarch64 termux-exec aarch64 1.4 [5516 B]
Get:9 https://termux.net stable/main aarch64 readline aarch64 8.2.13 [234 kB]
Get:10 https://termux.net stable/main aarch64 command-not-found aarch64 2.4.0-36 [80.3 kB]
Get:11 https://termux.net stable/main aarch64 debianutils aarch64 5.20 [16.2 kB]
Fetched 5365 kB in 2s (2695 kB/s)
(Reading database ... 3724 files and directories currently installed.)
Preparing to unpack .../archives/libc++_27_aarch64.deb ...
Unpacking libc++ (27) over (26d) ...
Setting up libc++ (27) ...
(Reading database ... 3725 files and directories currently installed.)
Preparing to unpack .../archives/apt_2.9.7_aarch64.deb ...
Unpacking apt (2.9.7) over (2.9.4-2) ...
Setting up apt (2.9.7) ...
(Reading database ... 3725 files and directories currently installed.)
Preparing to unpack .../ca-certificates_2024.07.02_all.deb ...
Unpacking ca-certificates (2024.07.02) over (2024.03.11-2) ...
Setting up ca-certificates (2024.07.02) ...
(Reading database ... 3725 files and directories currently installed.)
Preparing to unpack .../openssl_3.3.1_aarch64.deb ...
Unpacking openssl (3.3.1) over (3.2.1-1) ...
Setting up openssl (3.3.1) ...
(Reading database ... 3729 files and directories currently installed.)
Preparing to unpack .../libcurl_8.9.1_aarch64.deb ...
Unpacking libcurl (8.9.1) over (8.8.0) ...
Setting up libcurl (8.9.1) ...
(Reading database ... 3731 files and directories currently installed.)
Preparing to unpack .../curl_8.9.1_aarch64.deb ...
Unpacking curl (8.9.1) over (8.8.0) ...
Setting up curl (8.9.1) ...
(Reading database ... 3731 files and directories currently installed.)
Preparing to unpack .../archives/less_661_aarch64.deb ...
Unpacking less (661) over (633-1) ...
Setting up less (661) ...
(Reading database ... 3732 files and directories currently installed.)
Preparing to unpack .../termux-exec_1.4_aarch64.deb ...
Unpacking termux-exec (1.4) over (1.3) ...
Setting up termux-exec (1.4) ...
(Reading database ... 3732 files and directories currently installed.)
Preparing to unpack .../readline_8.2.13_aarch64.deb ...
Unpacking readline (8.2.13) over (8.2.10) ...
Setting up readline (8.2.13) ...
(Reading database ... 3732 files and directories currently installed.)
Preparing to unpack .../command-not-found_2.4.0-36_aarch64.deb ...
Unpacking command-not-found (2.4.0-36) over (2.4.0-30) ...
Preparing to unpack .../debianutils_5.20_aarch64.deb ...
Unpacking debianutils (5.20) over (5.19) ...
Setting up command-not-found (2.4.0-36) ...
Setting up debianutils (5.20) ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:  0     0    0     0    0     0      0      0 --:--:--  0:00:100     5  100     5    0     0      2      0  0:00:02  0:00:100     5  100     5    0     0      2      0  0:00:02  0:00:02 --:--:--     2
 ████████████████████████████████████████████

 ██╗░░░██╗██████╗░██████╗░░█████╗░██╗██████╗░
 ██║░░░██║██╔══██╗██╔══██╗██╔══██╗██║██╔══██╗
 ██║░░░██║██║░░██║██████╔╝██║░░██║██║██║░░██║
 ██║░░░██║██║░░██║██╔══██╗██║░░██║██║██║░░██║
 ╚██████╔╝██████╔╝██║░░██║╚█████╔╝██║██████╔╝
 ░╚═════╝░╚═════╝░╚═╝░░╚═╝░╚════╝░╚═╝╚═════╝░

 ████████████ (c)RandomCoderOrg ████████████
-> 78699 installs so far ....
-> Checking cpu architecture..
CPU architecture is OK.
-> Checking for required packages..
-> Installing required (missing) packages..
openssl is already the newest version (3.3.1).
Installing:
  git  jq  proot  pulseaudio  pv  wget

Installing dependencies:
  dbus                 libsndfile
  krb5                 libsoxr
  ldns                 libtalloc
  libandroid-execinfo  libuuid
  libdb                libvorbis
  libedit              libwebrtc-audio-processing
  libexpat             oniguruma
  libflac              openssh
  libltdl              openssh-sftp-server
  libogg               resolv-conf
  libopus              speexdsp
  libresolv-wrapper    termux-auth

Suggested packages:
  perl  proot-distro

Summary:
  Upgrading: 0, Installing: 30, Removing: 0, Not Upgrading: 0
  Download size: 11.2 MB
  Space needed: 59.2 MB

Get:1 https://termux.net stable/main aarch64 libexpat aarch64 2.6.2 [85.4 kB]
Get:2 https://termux.net stable/main aarch64 dbus aarch64 1.15.6-3 [425 kB]
Get:3 https://termux.net stable/main aarch64 git aarch64 2.46.0 [4505 kB]
Get:4 https://termux.net stable/main aarch64 oniguruma aarch64 6.9.9 [136 kB]
Get:5 https://termux.net stable/main aarch64 jq aarch64 1.7.1 [197 kB]
Get:6 https://termux.net stable/main aarch64 resolv-conf aarch64 1.3 [992 B]
Get:7 https://termux.net stable/main aarch64 libresolv-wrapper aarch64 1.1.7-4 [11.5 kB]
Get:8 https://termux.net stable/main aarch64 libdb aarch64 18.1.40-4 [511 kB]
Get:9 https://termux.net stable/main aarch64 krb5 aarch64 1.21.3 [900 kB]
Get:10 https://termux.net stable/main aarch64 ldns aarch64 1.8.3-2 [303 kB]
Get:11 https://termux.net stable/main aarch64 libandroid-execinfo aarch64 0.1-1 [14.8 kB]
Get:12 https://termux.net stable/main aarch64 libedit aarch64 20240517-3.1-0 [79.3 kB]
Get:13 https://termux.net stable/main aarch64 libogg aarch64 1.3.5 [189 kB]
Get:14 https://termux.net stable/main aarch64 libflac aarch64 1.4.3 [178 kB]
Get:15 https://termux.net stable/main aarch64 libltdl aarch64 2.4.7-4 [18.2 kB]
Get:16 https://termux.net stable/main aarch64 libopus aarch64 1.5.2 [186 kB]
Get:17 https://termux.net stable/main aarch64 libvorbis aarch64 1.3.7-2 [262 kB]
Get:18 https://termux.net stable/main aarch64 libsndfile aarch64 1.2.2 [218 kB]
Get:19 https://termux.net stable/main aarch64 libsoxr aarch64 0.1.3-6 [284 kB]
Get:20 https://termux.net stable/main aarch64 libtalloc aarch64 2.4.2 [33.1 kB]
Get:21 https://termux.net stable/main aarch64 libuuid aarch64 2.39.3-4 [14.6 kB]
Get:22 https://termux.net stable/main aarch64 libwebrtc-audio-processing aarch64 0.3.1-2 [269 kB]
Get:23 https://termux.net stable/main aarch64 openssh-sftp-server aarch64 9.8p1-4 [47.8 kB]
Get:24 https://termux.net stable/main aarch64 termux-auth aarch64 1.4-2 [6396 B]
Get:25 https://termux.net stable/main aarch64 openssh aarch64 9.8p1-4 [828 kB]
Get:26 https://termux.net stable/main aarch64 proot aarch64 5.1.107-66 [83.0 kB]
Get:27 https://termux.net stable/main aarch64 speexdsp aarch64 1.2.1 [37.4 kB]
Get:28 https://termux.net stable/main aarch64 pulseaudio aarch64 17.0-2 [843 kB]
Get:29 https://termux.net stable/main aarch64 pv aarch64 1.8.12 [282 kB]
Get:30 https://termux.net stable/main aarch64 wget aarch64 1.24.5 [287 kB]
Fetched 11.2 MB in 3s (4174 kB/s)
Selecting previously unselected package libexpat.
(Reading database ... 3732 files and directories currently installed.)
Preparing to unpack .../00-libexpat_2.6.2_aarch64.deb ...
Unpacking libexpat (2.6.2) ...
Selecting previously unselected package dbus.
Preparing to unpack .../01-dbus_1.15.6-3_aarch64.deb ...
Unpacking dbus (1.15.6-3) ...
Selecting previously unselected package git.
Preparing to unpack .../02-git_2.46.0_aarch64.deb ...
Unpacking git (2.46.0) ...
Selecting previously unselected package oniguruma.
Preparing to unpack .../03-oniguruma_6.9.9_aarch64.deb ...
Unpacking oniguruma (6.9.9) ...
Selecting previously unselected package jq.
Preparing to unpack .../04-jq_1.7.1_aarch64.deb ...
Unpacking jq (1.7.1) ...
Selecting previously unselected package resolv-conf.
Preparing to unpack .../05-resolv-conf_1.3_aarch64.deb ...
Unpacking resolv-conf (1.3) ...
Selecting previously unselected package libresolv-wrapper.
Preparing to unpack .../06-libresolv-wrapper_1.1.7-4_aarch64.deb ...
Unpacking libresolv-wrapper (1.1.7-4) ...
Selecting previously unselected package libdb.
Preparing to unpack .../07-libdb_18.1.40-4_aarch64.deb ...
Unpacking libdb (18.1.40-4) ...
Selecting previously unselected package krb5.
Preparing to unpack .../08-krb5_1.21.3_aarch64.deb ...
Unpacking krb5 (1.21.3) ...
Selecting previously unselected package ldns.
Preparing to unpack .../09-ldns_1.8.3-2_aarch64.deb ...
Unpacking ldns (1.8.3-2) ...
Selecting previously unselected package libandroid-execinfo.
Preparing to unpack .../10-libandroid-execinfo_0.1-1_aarch64.deb ...
Unpacking libandroid-execinfo (0.1-1) ...
Selecting previously unselected package libedit.
Preparing to unpack .../11-libedit_20240517-3.1-0_aarch64.deb ...
Unpacking libedit (20240517-3.1-0) ...
Selecting previously unselected package libogg.
Preparing to unpack .../12-libogg_1.3.5_aarch64.deb ...
Unpacking libogg (1.3.5) ...
Selecting previously unselected package libflac.
Preparing to unpack .../13-libflac_1.4.3_aarch64.deb ...
Unpacking libflac (1.4.3) ...
Selecting previously unselected package libltdl.
Preparing to unpack .../14-libltdl_2.4.7-4_aarch64.deb ...
Unpacking libltdl (2.4.7-4) ...
Selecting previously unselected package libopus.
Preparing to unpack .../15-libopus_1.5.2_aarch64.deb ...
Unpacking libopus (1.5.2) ...
Selecting previously unselected package libvorbis.
Preparing to unpack .../16-libvorbis_1.3.7-2_aarch64.deb ...
Unpacking libvorbis (1.3.7-2) ...
Selecting previously unselected package libsndfile.
Preparing to unpack .../17-libsndfile_1.2.2_aarch64.deb ...
Unpacking libsndfile (1.2.2) ...
Selecting previously unselected package libsoxr.
Preparing to unpack .../18-libsoxr_0.1.3-6_aarch64.deb ...
Unpacking libsoxr (0.1.3-6) ...
Selecting previously unselected package libtalloc.
Preparing to unpack .../19-libtalloc_2.4.2_aarch64.deb ...
Unpacking libtalloc (2.4.2) ...
Selecting previously unselected package libuuid.
Preparing to unpack .../20-libuuid_2.39.3-4_aarch64.deb ...
Unpacking libuuid (2.39.3-4) ...
Selecting previously unselected package libwebrtc-audio-processing.
Preparing to unpack .../21-libwebrtc-audio-processing_0.3.1-2_aarch64.deb ...
Unpacking libwebrtc-audio-processing (0.3.1-2) ...
Selecting previously unselected package openssh-sftp-server.
Preparing to unpack .../22-openssh-sftp-server_9.8p1-4_aarch64.deb ...
Unpacking openssh-sftp-server (9.8p1-4) ...
Selecting previously unselected package termux-auth.
Preparing to unpack .../23-termux-auth_1.4-2_aarch64.deb ...
Unpacking termux-auth (1.4-2) ...
Selecting previously unselected package openssh.
Preparing to unpack .../24-openssh_9.8p1-4_aarch64.deb ...
Unpacking openssh (9.8p1-4) ...
Selecting previously unselected package proot.
Preparing to unpack .../25-proot_5.1.107-66_aarch64.deb ...
Unpacking proot (5.1.107-66) ...
Selecting previously unselected package speexdsp.
Preparing to unpack .../26-speexdsp_1.2.1_aarch64.deb ...
Unpacking speexdsp (1.2.1) ...
Selecting previously unselected package pulseaudio.
Preparing to unpack .../27-pulseaudio_17.0-2_aarch64.deb ...
Unpacking pulseaudio (17.0-2) ...
Selecting previously unselected package pv.
Preparing to unpack .../28-pv_1.8.12_aarch64.deb ...
Unpacking pv (1.8.12) ...
Selecting previously unselected package wget.
Preparing to unpack .../29-wget_1.24.5_aarch64.deb ...
Unpacking wget (1.24.5) ...
Setting up libopus (1.5.2) ...
Setting up libedit (20240517-3.1-0) ...
Setting up openssh-sftp-server (9.8p1-4) ...
Setting up libuuid (2.39.3-4) ...
Setting up resolv-conf (1.3) ...
Setting up wget (1.24.5) ...
Setting up libsoxr (0.1.3-6) ...
Setting up ldns (1.8.3-2) ...
Setting up libogg (1.3.5) ...
Setting up speexdsp (1.2.1) ...
Setting up libexpat (2.6.2) ...
Setting up libvorbis (1.3.7-2) ...
Setting up libandroid-execinfo (0.1-1) ...
Setting up libltdl (2.4.7-4) ...
Setting up dbus (1.15.6-3) ...
Setting up git (2.46.0) ...
Setting up libwebrtc-audio-processing (0.3.1-2) ...
Setting up pv (1.8.12) ...
Setting up libresolv-wrapper (1.1.7-4) ...
Setting up libtalloc (2.4.2) ...
Setting up termux-auth (1.4-2) ...
Setting up libdb (18.1.40-4) ...
Setting up libflac (1.4.3) ...
Setting up oniguruma (6.9.9) ...
Setting up proot (5.1.107-66) ...
Setting up libsndfile (1.2.2) ...
Setting up krb5 (1.21.3) ...
Setting up jq (1.7.1) ...
Setting up openssh (9.8p1-4) ...
Generating public/private rsa key pair.
Your identification has been saved in /data/data/com.termux/files/usr/etc/ssh/ssh_host_rsa_key
Your public key has been saved in /data/data/com.termux/files/usr/etc/ssh/ssh_host_rsa_key.pub
The key fingerprint is:
SHA256:IrPjEiGKu5Oi2iAoqHmOdYg4bYPOZKjaVZWwhhXcrOQ u0_a227@localhost
The key's randomart image is:
+---[RSA 3072]----+
|     .++         |
|     oooo.       |
|    .oo.o        |
|. .  .E.         |
|o. .o o S        |
|B+.. = .         |
|%=*.=            |
|#Xo= .           |
|#B+..            |
+----[SHA256]-----+
unknown key type dsa
Generating public/private ecdsa key pair.
Your identification has been saved in /data/data/com.termux/files/usr/etc/ssh/ssh_host_ecdsa_key
Your public key has been saved in /data/data/com.termux/files/usr/etc/ssh/ssh_host_ecdsa_key.pub
The key fingerprint is:
SHA256:0hd3dL8YenUBxviKI4YQaP7FSHm0S69na+Id8T/s0f8 u0_a227@localhost
The key's randomart image is:
+---[ECDSA 256]---+
|  . o.      ooo..|
| o + ..    ..o .o|
|o . =o    . o...o|
| . o.oo.   o.o+ o|
|  . o.ooS o..o . |
|   . ..+o+ .o    |
|     ..+..o. .   |
|     .+.o .o. .  |
|    ..oo  .o.  .E|
+----[SHA256]-----+
Generating public/private ed25519 key pair.
Your identification has been saved in /data/data/com.termux/files/usr/etc/ssh/ssh_host_ed25519_key
Your public key has been saved in /data/data/com.termux/files/usr/etc/ssh/ssh_host_ed25519_key.pub
The key fingerprint is:
SHA256:aMS5go/kdNxH2Xj/5s8dxEN5DlW4/kOZ+Rg14xX7XHU u0_a227@localhost
The key's randomart image is:
+--[ED25519 256]--+
|               .+|
|     . . +    .oE|
|      + + o   .+*|
|   o o + . .  +*=|
|  + + = S   ...*@|
| + + o .     .+*+|
|  o .         +=.|
|             o.o=|
|              ..=|
+----[SHA256]-----+
Setting up pulseaudio (17.0-2) ...
Cloning into 'fs-manager-udroid'...
remote: Enumerating objects: 2044, done.
remote: Counting objects: 100% (500/500), done.
remote: Compressing objects: 100% (101/101), done.
remote: Total 2044 (delta 415), reused 455 (delta 395), pack-reused 1544
Receiving objects: 100% (2044/2044), 14.86 MiB | 2.58 MiB/s, done.
Resolving deltas: 100% (1012/1012), done.
jq is already the newest version (1.7.1).
wget is already the newest version (1.24.5).
proot is already the newest version (5.1.107-66).
pv is already the newest version (1.8.12).
pulseaudio is already the newest version (17.0-2).
Installing:
  libandroid-shmem-static  which

Installing dependencies:
  libandroid-shmem

Summary:
  Upgrading: 0, Installing: 3, Removing: 0, Not Upgrading: 0
  Download size: 25.2 kB
  Space needed: 209 kB

Get:1 https://termux.net stable/main aarch64 libandroid-shmem aarch64 0.5 [6936 B]
Get:2 https://termux.net stable/main aarch64 libandroid-shmem-static aarch64 0.5 [5768 B]
Get:3 https://termux.net stable/main aarch64 which aarch64 2.21 [12.5 kB]
Fetched 25.2 kB in 2s (13.8 kB/s)
Selecting previously unselected package libandroid-shmem.
(Reading database ... 5771 files and directories currently installed.)
Preparing to unpack .../libandroid-shmem_0.5_aarch64.deb ...
Unpacking libandroid-shmem (0.5) ...
Selecting previously unselected package libandroid-shmem-static.
Preparing to unpack .../libandroid-shmem-static_0.5_aarch64.deb ...
Unpacking libandroid-shmem-static (0.5) ...
Selecting previously unselected package which.
Preparing to unpack .../which_2.21_aarch64.deb ...
Unpacking which (2.21) ...
Setting up libandroid-shmem (0.5) ...
Setting up which (2.21) ...
Setting up libandroid-shmem-static (0.5) ...

[Warning]: Android version 15 detected
You may experience issues like crashing
watch this to fix signal 9 issue:
      https://youtu.be/GCN0gh1yXSs?si=ofVGAuDGKS1mMbS1


/data/data/com.termux/files/usr/bin/which: no udroid in (/data/data/com.termux/files/usr/bin:/product/bin:/apex/com.android.runtime/bin:/apex/com.android.art/bin:/system_ext/bin:/system/bin:/system/xbin:/odm/bin:/vendor/bin:/vendor/xbin)
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] Creating directory "/data/data/com.termux/files/usr/etc/udroid"...
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] Creating directory "/data/data/com.termux/files/usr/var/lib/udroid"...
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] Creating directory "/data/data/com.termux/files/usr/var/lib/udroid/installed-filesystems"...
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] Creating directory "/data/data/com.termux/files/usr/var/lib/udroid/dlcache"...
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] Creating directory "/data/data/com.termux/files/usr/etc/udroid/.cache"...
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] installing src...
error: "/data/data/com.termux/files/home/fs-manager-udroid/udroid/src/gum/gum-aarch64/usr/bin/gum" has unexpected e_type: 2
[✔] processing symbolic link...
[-] Installation complete
-> Installing jammy:xfce4 in 3 seconds..
-> press q or CTRL+C to stop here..
No command udroid found, did you mean:
 Command drop4 in package bsd-games
 Command crond in package cronie
 Command fdroidcl in package fdroidcl
 Command kpropd in package krb5
 Command uuid in package ossp-uuid
 Command dropdb in package postgresql
[E] Failed to install jammy:xfce4
-> Installation completed successfully.
->
-> To login jammy:xfce4, run "udroid login jammy:xfce4"
-> use vncserver :1 to start vnc server
-> default password for everything is secret
-> Show us some love by starring our repo on github and donating

ubuntu-on-android: https://github.com/RandomCoderOrg/ubuntu-on-android
support us:        https://github.com/sponsors/RandomCoderOrg
Join our discord:  https://discord.gg/AGqQCHuE6S

~/fs-manager-udroid
```

**Comment by SaicharanKandukuri** [comment]
~~I can't reproduce this issue~~, it's probably related to this issue
 - https://github.com/termux/termux-exec/pull/24

> Issue is with the udroid dependency [`gum`](https://github.com/charmbracelet/gum) used for terminal prompts.

---

> https://github.com/RandomCoderOrg/fs-manager-udroid/issues/54#issuecomment-2278594464


Seems like this issue happens only on the termux playstore version and [F-Droid](https://f-droid.org/en/packages/com.termux/) version is fine. 

Until this issue is fixed from the termux side udroid should have a failsafe to work without gum static binaries.

**Comment by SaicharanKandukuri** [comment]
Issues closes after RandomCoderOrg/fs-manager-udroid#54

**Comment by SaicharanKandukuri** [comment]
Closing for now
